src/devices/Card/ProcessError.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Iot.Device.Card
+{
+    /// <summary>
+    /// Processing error class
+    /// </summary>
+    public class ProcessError
+    {
+        /// <summary>
+        /// The Error type
+        /// </summary>
+        public ErrorType ErrorType { get; internal set; }
+
+        /// <summary>
+        /// Complementary data for some errors
+        /// </summary>
+        public byte CorrectLegnthOrBytesAvailable { get; internal set; }
+
+        /// <summary>
+        /// Constructor to process the error
+        /// </summary>
+        /// <param name="errorToProcess">A span of byte</param>
+        public ProcessError(Span<byte> errorToProcess)
+        {
+            // EMV 4.3 Book 3 page 60
+            if (errorToProcess.Length < 2)
+                ErrorType = ErrorType.Unknown;
+            else if ((errorToProcess[0] == 0x90) && (errorToProcess[1] == 0x00))
```

Comment:
this piece of code feels like it could be collapsed to something like:

`(ErrorType)(errorToProcess[0] | (errorToProcess[1] << 8))` + changes to `ErrorType`

Reply:
There are paths for errors. And some errors are pure errors, some errors are not pure error, the second byte contains a complementary information. They are grouped by family which can be identified with the first byte. See a quite good and exhaustive list of possible errors, warnings and incorrect data lengths, here: https://www.eftlab.co.uk/knowledge-base/complete-list-of-apdu-responses/